manager/snap.go

```diff
@@ -79,6 +87,126 @@ func (snap *snap) GetProxySettings() (proxy.Settings, error) {
 	return res, nil
 }
 
+// ConfigureStoreProxyFromURL downloads an assertion list from the snap store
+// proxy located at proxyURL and sets up snapd to use the proxy and store ID
+// obtained from the retrieved assertions.
+//
+// If the local snap store proxy instance is operating in an air-gapped
+// environment, downloading the assertion list from the proxy will not be
+// possible and an appropriate error will be returned. For such use-cases, the
+// store proxy must be configured via a call to ConfigureStoreProxy instead.
+//
+// If snap also needs to use HTTP/HTTPS proxies to talk to the outside world,
+// these need to be configured separately before invoking this method via a
+// call to SetProxy.
+func (snap *Snap) ConfigureStoreProxyFromURL(proxyURL string) error {
+	u, err := url.Parse(proxyURL)
+	if err != nil {
+		return errors.Annotate(err, "proxy URL not valid")
+	}
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return errors.NotValidf("proxy URL scheme %q", u.Scheme)
+	}
+
+	res, err := http.Get(proxyURL + "/v2/auth/store/assertions")
+	if err != nil {
+		return errors.Annotatef(err, "could not contact snap store proxy at %q", proxyURL)
+	}
+	defer func() { _ = res.Body.Close() }()
+	if res.StatusCode != http.StatusOK {
+		return errors.Annotatef(err, "could not retrieve assertions from proxy at %q. If using an air-gapped proxy you must manually provide the assertions file and store ID", proxyURL)
```

Comment:
I'm unsure this is the best way to find out if we're air gapped, but if there is no other option can we set a time before using the default `http.Get`.

```go
client := http.Client{
	Timeout:  time.Duration(30 * time.Second),
}
res, err := client.Get(url)
```